plugins/aws/sam.go

```diff
@@ -0,0 +1,26 @@
+package aws
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func AWSSAMCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "AWS SAM CLI",
+		Runs:    []string{"sam"},
+		DocsURL: sdk.URL("https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/reference-sam-cli.html"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
```

Comment:
I believe there are a few more commands that do not require Auth, such as `sam validate`. 

I managed to run this command via `AWS_DEFAULT_REGION=us-west-2 sam validate` with both valid and invalid templates and get the expected results.

Similarly for `sam init` - can you please have a look at all the commands that should be omitted and add them here?

Reply:
Thanks for the suggestion @hculea - done!